src/main/java/io/github/jhipster/registry/service/CloudConfigRefreshService.java

```diff
@@ -0,0 +1,181 @@
+                    thread.setDaemon(true);
+                    return thread;
+                }
+            );
+            taskExecutor.execute(() -> {
+                try {
+                    configMapRefreshContext();
+                } catch (IOException | InterruptedException ex) {
+                    log.error("Unable to refresh K8s ConfigMap", ex);
+                }
+            });
+        } else {
+            log.error("ConfigMap directory path not specified. Specify value for the environment variable k8s.config.path");
+        }
+    }
+
+    /**
+     * {@code WatchService} object to monitor K8s configMap path. Mounted configMap path will be recursively
+     * registered with the {@code WatchService} instance to get notified for interested events.
+     *
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public void configMapRefreshContext() throws IOException, InterruptedException {
+        List<File> fileList = new ArrayList();
+        List<Integer> hashList = new ArrayList();
+        WatchService watcherService = FileSystems.getDefault().newWatchService();
+        Path dirPath = Paths.get(getConfigPath());
+        Files.walkFileTree(dirPath, new HashSet<FileVisitOption>() {
+            {
+                add(FOLLOW_LINKS);
+            }
+        }, 2, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                log.debug("Registering" + dir + " in watcher service");
+                dir.register(watcherService, new WatchEvent.Kind[]{ENTRY_MODIFY}, HIGH);
+                return CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult visitFile(Path path, BasicFileAttributes attrs) throws IOException {
+                File file = path.toFile();
+                if (isValidConfigFile(file.getName().toLowerCase())) {
+                    log.debug("Adding file: " + file.getAbsolutePath());
+                    fileList.add(file);
+                    hashList.add(getHashValue(file));
+                }
+                return CONTINUE;
+            }
+        });
+        while (true) {
+            WatchKey key = watcherService.take();
+            List<WatchEvent<?>> events = key.pollEvents();
+            if (!events.isEmpty()) {
+                if (log.isDebugEnabled()) {
+                    events.forEach(event -> log.debug("Event detected: " + event.kind().name() + ", Updated File: " + event.context()));
+                }
+                Collection<Integer> activeList = fileList.stream().map(entry -> getHashValue(entry)).collect(Collectors.toList());
+                if (!hashList.containsAll(activeList)) {
+                    log.debug("File system updated. Hashed content matching failed");
+                    hashList.clear();
+                    hashList.addAll(activeList);
+                    refresher.refresh();
```

Comment:
This would refresh only the registry instances (@RefreshScope beans), and not the registered apps. In order to auto refresh the registered apps on repo update, we need Kafka (or RabbitMQ/Redis) to trigger ContextRefresher.refresh() or RefreshEndpoint.refresh() (which is similar to /refresh POST call)